src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -1613,12 +1819,32 @@ def impute_timeseries_asset_factory(
         reported_value_col: Name of column in output asset with original reported values.
         id_col: Name of column identifying entities to group timeseries by.
         output_io_manager_key: IO-manager to use for final output asset.
+        real_id_cols: The imputation process requires a single ID per timeseries', but some
+            tables like EIA 930 subregion demand actually have two ID columns. We use
+            a temp combined ID in these cases, but this temp column gets dropped
+            on the output asset, which is used for scoring our simulated imputation.
+            If specified, these columns will be used instead during scoring.
+        settings: Configurable options for imputation (see :class:`ImputeTimeseriesSettings`).
     """
-    timeseries_matrix_asset = f"_{output_asset_name}_timeseries_matrix"
-    aligned_input_asset = f"_{input_asset_name}_aligned"
-    cleaned_timeseries_matrix_asset = f"_{output_asset_name}_cleaned_timeseries_matrix"
-    flags_asset = f"_{output_asset_name}_timeseries_matrix_flags"
-    imputed_asset = f"_{output_asset_name}_imputed_matrix"
+    timeseries_matrix_asset = re.sub(
+        r"^__", "_", f"_{output_asset_name}_timeseries_matrix"
```

Comment:
Can you add a comment explaining what's up with all the regex substititions here? I think I know, but it's confusing at first.

Reply:
Added, it's basically just making sure these assets only have one underscore